qutip/tests/core/data/test_mathematics.py

```diff
@@ -775,6 +786,44 @@ def test_incorrect_shape_raises(self, op, data_m):
             op(data_m())
 
 
+class TestPow(UnaryOpMixin):
+    def op_numpy(self, matrix, n):
+        return np.linalg.matrix_power(matrix, n)
+
+    shapes = shapes_square()
+    bad_shapes = shapes_not_square()
+    specialisations = [
+        pytest.param(data.pow_csr, CSR, CSR),
+    ]
+
+    @pytest.mark.parametrize("n", [1, 10], ids=["n_1", "n_10"])
+    def test_mathematically_correct(self, op, data_m, out_type, n):
+        matrix = data_m()
+        expected = self.op_numpy(matrix.to_array(), n)
+        test = op(matrix, n)
+        assert isinstance(test, out_type)
+        if issubclass(out_type, Data):
+            assert test.shape == expected.shape
+            np.testing.assert_allclose(test.to_array(), expected,
+                                       atol=self.tol)
+        elif out_type is list:
+            for test_, expected_ in zip(test, expected):
+                assert test_.shape == expected_.shape
+                np.testing.assert_allclose(test_.to_array(),
+                                           expected_, atol=self.tol)
+        else:
+            assert abs(test - expected) < self.tol
```

Comment:
We know that the output is a `Data` object. 
```suggestion
        assert test.shape == expected.shape
        np.testing.assert_allclose(test.to_array(), expected,
                                   atol=self.tol)
```